ProductHeaderValue parser throws exception for empty values

## Steps to Reproduce

1. Create a project targeting .NET framework on a MacOS machine
2. Use the snippet below to create a `.cs` file
3. Restore, Build and run the exe using mono

```csharp
using System.Net.Http;
            var pihv = new ProductInfoHeaderValue("OSNAME", "");
            System.Console.WriteLine(pihv.Product+"=>"+pihv.Comment);
            Console.ReadKey();
```

### Current Behavior
The `ProductHeaderValue` ctor throws an exception in the parser an empty `value` is passed to it
The Exception comes from [here](https://github.com/mono/mono/blob/d36902effdcb02d93b14943ce3ceaf1bcf11619a/mcs/class/System.Net.Http/System.Net.Http.Headers/Parser.cs#L57)

### Expected Behavior

The ProductInfoHeaderValue object should be created successfully with an empty value. The same code on a windows implementation of System.Net.Http works as expected without throwing any exceptions

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

mono version: 
```
Mono JIT compiler version 5.10.1.57 (2017-12/ea8a24b1bbf Tue Apr 24 14:53:01 EDT 2018)
```

## Stacktrace

```
Unhandled Exception:
System.ArgumentException: Value does not fall within the expected range.
  at System.Net.Http.Headers.Parser+Token.Check (System.String s) [0x00019] in <424e46dead96486786256709cc4c63d9>:0 
  at System.Net.Http.Headers.ProductHeaderValue..ctor (System.String name, System.String version) [0x0000a] in <424e46dead96486786256709cc4c63d9>:0 
  at System.Net.Http.Headers.ProductInfoHeaderValue..ctor (System.String productName, System.String productVersion) [0x00006] in <424e46dead96486786256709cc4c63d9>:0 
  at sctest.Program.Main (System.String[] args) [0x00001] in <b76927fa3c6447ed98c8230a28e82d3b>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: Value does not fall within the expected range.
  at System.Net.Http.Headers.Parser+Token.Check (System.String s) [0x00019] in <424e46dead96486786256709cc4c63d9>:0 
  at System.Net.Http.Headers.ProductHeaderValue..ctor (System.String name, System.String version) [0x0000a] in <424e46dead96486786256709cc4c63d9>:0 
  at System.Net.Http.Headers.ProductInfoHeaderValue..ctor (System.String productName, System.String productVersion) [0x00006] in <424e46dead96486786256709cc4c63d9>:0 
  at sctest.Program.Main (System.String[] args) [0x00001] in <b76927fa3c6447ed98c8230a28e82d3b>:0 
 

```

--- a/mcs/class/System.Net.Http/System.Net.Http.Headers/ProductHeaderValue.cs
+++ b/mcs/class/System.Net.Http/System.Net.Http.Headers/ProductHeaderValue.cs
@@ -41,7 +41,7 @@ namespace System.Net.Http.Headers
 		public ProductHeaderValue (string name, string version)
 			: this (name)
 		{
-			if (version != null)
+			if (!string.IsNullOrEmpty (version))
 				Parser.Token.Check (version);
 
 			Version = version;

--- a/mcs/class/System.Net.Http/Test/System.Net.Http.Headers/ProductHeaderValueTest.cs
+++ b/mcs/class/System.Net.Http/Test/System.Net.Http.Headers/ProductHeaderValueTest.cs
@@ -63,6 +63,7 @@ namespace MonoTests.System.Net.Http.Headers
 		public void Ctor ()
 		{
 			new ProductHeaderValue ("aa", null);
+			new ProductHeaderValue ("aa", "");
 		}
 
 		[Test]

